commit dc04a00b3cbb89c96e3b5d5ef7b7cb75d1d747ec
Author: Alexandre Ghiti <aghiti@upmem.com>
Date:   Mon Dec 10 06:21:46 2018 +0000

    riscv: Adjust mmap base address at a third of task size
    
    [ Upstream commit ae662eec8a515ab550524e04c793b5ddf1aae3a1 ]
    
    This ratio is the most used among all other architectures and make
    icache_hygiene libhugetlbfs test pass: this test mmap lots of
    hugepages whose addresses, without this patch, reach the end of
    the process user address space.
    
    Signed-off-by: Alexandre Ghiti <aghiti@upmem.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Palmer Dabbelt <palmer@sifive.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/arch/riscv/include/asm/processor.h b/arch/riscv/include/asm/processor.h
index 3fe4af8147d2..c23578a37b44 100644
--- a/arch/riscv/include/asm/processor.h
+++ b/arch/riscv/include/asm/processor.h
@@ -22,7 +22,7 @@
  * This decides where the kernel will search for a free chunk of vm
  * space during mmap's.
  */
-#define TASK_UNMAPPED_BASE	PAGE_ALIGN(TASK_SIZE >> 1)
+#define TASK_UNMAPPED_BASE	PAGE_ALIGN(TASK_SIZE / 3)
 
 #define STACK_TOP		TASK_SIZE
 #define STACK_TOP_MAX		STACK_TOP
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1015_dc04a00b3cbb89c96e3b5d5ef7b7cb75d1d747ec.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1015_dc04a00b3cbb89c96e3b5d5ef7b7cb75d1d747ec.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
